cmd/fossa/cmd/analyze/analyze.go

```diff
@@ -133,6 +133,68 @@ func Do(modules []module.Module) (analyzed []module.Module, err error) {
 }
 
 func uploadAnalysis(normalized []fossa.SourceUnit) error {
+	supportedApiVersions, err := fossa.GetSupportedApiVersions()
+	if err != nil {
+		return err
+	}
+
+	// lists out all compatible versions, where versions listed earlier in the list should be preferred to be used over later appearing versions
+	compatibleVersionsByPreference := []string{"v1", "v0"}
+	for _, compatibleVersion := range compatibleVersionsByPreference {
```

Comment:
This code is pretty crazy, and we don't need it if we change the interface to `fossa.GetLatestSupportedAPIVersion()`.

Reply:
I agree. It looks like we are looping through `compatibleVersionsByPreference` and then later switching on the same values. Is there a reason to check for a supported version twice?

Reply:
I'd be okay with moving this to be scoped within the fossa API package. 

My goal in having this loop was to handle situations where users have an older version of `cli`   or on prem users that may have an older version of core.

@zlav Its not checking supported version twice, it's determining which versions that core supports are also supported by this build of `cli`